lib/models/device.coffee

```diff
@@ -1268,6 +1268,36 @@ getDeviceModel = (deps, opts) ->
 		.asCallback(callback)
 
 	###*
+	# @summary Generate a device key
+	# @name generateDeviceKey
+	# @public
+	# @function
+	# @memberof resin.models.device
+	#
+	# @param {String|Number} uuidOrId - device uuid (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.device.generateDeviceKey('7cf02a6');
+	#
+	# @example
+	# resin.models.device.generateDeviceKey(123);
+	#
+	# @example
+	# resin.models.device.generateDeviceKey('7cf02a6', function(error) {
+	# 	if (error) throw error;
+	# });
+	###
+	exports.generateDeviceKey = (uuidOrId, callback) ->
+		getId(uuidOrId).then (deviceId) ->
+			return request.send
+				method: 'POST'
+				url: "/api-key/device/#{deviceId}/device-key"
+				baseUrl: apiUrl
```

Comment:
What happens if you pass an id, and the device doesn't exist?

`getId` won't throw, it'll just pass it through. Assuming that endpoint comes back with a 404, you'll just get a `ResinRequestError`. It'd be better to do `.catch(notFoundResponse, treatAsMissingDevice(uuidOrId))`.